src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,282 @@
+package duke;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.exceptions.DukeDateTimeParseException;
+import duke.exceptions.DukeDeadlineFormatException;
+import duke.exceptions.DukeEmptyActionException;
+import duke.exceptions.DukeEmptyDeadlineException;
+import duke.exceptions.DukeEmptyEventException;
+import duke.exceptions.DukeEmptyFindException;
+import duke.exceptions.DukeEmptyToDoException;
+import duke.exceptions.DukeEventFormatException;
+import duke.exceptions.DukeException;
+import duke.exceptions.DukeInvalidIndexException;
+import duke.exceptions.DukeInvalidTaskException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+
+
+/**
+ * Parser deals with making sense of the user command.
+ */
+public class Parser {
+    private String commandLine;
+
+    /**
+     * Parser constructor.
+     * @param commandLine The string of text given by the user.
+     */
+    Parser(String commandLine) {
+        this.commandLine = commandLine;
+    }
+
+    /**
+     * The main method that analyse the user input and give an
+     * appropriate command.
```

Comment:
Should this method header comment start with "Analyses.." instead?

Reply:
I noticed the same issue in several other places too.

Reply:
Thank you for pointing this out!